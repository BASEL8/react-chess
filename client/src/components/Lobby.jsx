import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import ClearIcon from '@material-ui/icons/Clear';

const useStyles = makeStyles({
  wrapper: {
    diplay: 'flex',
    justifyContent: 'flex-start',
    flexWrap: 'nowrap'
  },
  table: {
    maxWidth: '60vw',
  },
  buttonContainer: {
    maxWidth: '25vw',
    display: 'flex',
    flexDirection: 'column',
    alignContent: 'center'
  }
});

function createData(id, player, fat, carbs, protein) {
  return { id, player, fat, carbs, protein };
}

const rows = [
  createData(1, 'Frozen yoghurt', 6.0, 24),
  createData(2, 'Ice cream sandwich', 9.0, 37),
  createData(3, 'Eclair', 16.0, 24),
  createData(4, 'Cupcake', 3.7, 67),
  createData(5, 'Gingerbread', 16.0, 49),
];

export default function Lobby () {
  const classes = useStyles();

  return (
    <div className={classes.wrapper}>
      <TableContainer component={Paper} className={classes.table}>
        <Table aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>No.</TableCell>
              <TableCell>Player</TableCell>
              <TableCell>Play</TableCell>
              <TableCell>Delete</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map(row => (
              <TableRow key={row.id}>
                <TableCell component="th" scope="row">
                  {row.id}
                </TableCell>
                <TableCell>{row.player}</TableCell>
                <TableCell><Button variant="outlined">Play</Button></TableCell>
                <TableCell><Button><ClearIcon /></Button></TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
      <div className={classes.buttonContainer}>
        <Button variant="outlined">Create a new match</Button>
        <Button variant="outlined">Play with a friend</Button>
        <Button variant="outlined">Play against computer</Button>
      </div>
    </div>
  );
}
