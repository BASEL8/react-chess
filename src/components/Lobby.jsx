/* eslint-disable no-undef */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Helmet from 'react-helmet';

const Lobby = () => {
  const [seeks, setSeeks] = useState([]);

  useEffect(() => {
    axios.get(`http://localhost:8000/api/seeks/${localStorage.getItem('userId')}`)
      .then((response) => {
        // console.log(response.data);
        setSeeks(response.data);
      });
  }, []);

  if (!seeks) {
    return <p>Loading seeks...</p>;
  }

  return (
    <>
      <Helmet><title>Lobby</title></Helmet>
      <table>
        <thead>
          <tr>
            <th>Player One</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {seeks.map((seek) => (
            <>
              <tr key={seek.id}>
                <td>{seek.playerOne}</td>
                <td>
                  <button type="submit">Play</button>
                  {' '}
                </td>
              </tr>
            </>
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Lobby;
