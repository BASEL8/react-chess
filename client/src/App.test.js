import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import Lobby from './components/Lobby';
import { ExpansionPanelActions } from '@material-ui/core';

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Lobby text="Hello!" />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders table", () => {
  const { getByTestId, getByText } = render(<Lobby text="Hello!" />);
  expect(getByTestId('table')).toHaveTextContent('Hello!');
})