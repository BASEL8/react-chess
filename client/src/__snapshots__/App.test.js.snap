// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrapper-1"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiTableContainer-root makeStyles-table-2 MuiPaper-rounded"
    >
      <table
        aria-label="simple table"
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              No.
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Player
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Play
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Delete
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              1
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Frozen yoghurt
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Play
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              2
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Ice cream sandwich
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Play
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              3
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Eclair
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Play
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              4
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Cupcake
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Play
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              5
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Gingerbread
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Play
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="makeStyles-buttonContainer-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Create a new match
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Play with a friend
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Play against computer
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
